Core/include/Acts/Utilities/DelegateChainBuilder.hpp

```diff
@@ -0,0 +1,161 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Delegate.hpp"
+#include "Acts/Utilities/TypeList.hpp"
+#include "Acts/Utilities/TypeTag.hpp"
+
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace Acts {
+
+template <typename, typename payload_types = TypeList<>, auto... Callables>
+class DelegateChainBuilder;
+
+// @TODO: Maybe add concept requirement for default initialization of R
```

Comment:
todo for now or later?

Reply:
I had to do this in a slightly awkward way due to how the function type is decomposed using template specialization. It seems I can't add a regular class requirements, because that would differ between specializations. I added a `static_assert` now that make the error much shorter and provides an error message, rather than an implicit error further down.